lib/improver/threshold.py

```diff
@@ -201,13 +200,13 @@ def _add_threshold_coord(self, cube, threshold):
         try:
             coord = iris.coords.DimCoord(
                 np.array([threshold], dtype=np.float32),
-                standard_name=self.threshold_coord_name,
+                standard_name=cube.name(),
```

Comment:
Um guys this isn't safe.  You are assuming we have never renamed the cube before creating this threshold coordinate, and you haven't commented this as a MASSIVE RED WARNING in the docstring - which makes this code vulnerable to further developments that break it in non-obvious ways.  There are functions to do this in the IMPROVER code base, and checks to make sure the cube hasn't been renamed - but this implementation is not right.